wordpress-mobile/WordPress-Android
# [Site Creation] Step 1: Remove vertical id in site creation

This PR stops sending vertical id to new site requests as part of #11420 and #11421 and completes vertical step removal from the site creation flow.

PR in FluxC - https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1511

P.S. Review after #11435 is merged to the Master PR.

**To test:** 

- Go to My site
- Click Switch Site
- Click "+" button to launch Add new site
- Click Create WordPress.com site on Add new site
- Choose a segment Blog/Business/Professional/Blank Canvas to open Step 1
- Add Site Title on Site info step displayed as Step 1 of 2 
- Click Next to open domains step (Step 2 of 2)
- Select a domain from auto searched results
- Click Create Site button to create a new site
- Verify  new site is created successfully and `verticalId` is not sent as `site_vertical` in options parameter in `/sites/new` request from Stetho
<img width="779" alt="new_site" src="https://user-images.githubusercontent.com/1405144/76732075-d1b70d00-6784-11ea-8fa0-241eb1ce50e0.png">

[Click OK on "Your site has been created" and delete above test site from My Site -> Settings.]

PR submission checklist:
